src/utils/junitUtils.ts

```diff
@@ -0,0 +1,40 @@
+import { SendJsonOutput } from '@/cli/send/jsonOutput';
+
+export function toJunitXml(output: SendJsonOutput): string {
+  let resultXml = `
+  <?xml version="1.0" encoding="UTF-8"?>
+  <testsuites 
+    disabled="0"
+    errors="${output.requests.flatMap(r => r.testResults).filter(r => r?.error !== undefined).length}"
+    failures="${output.summary.failedTests}"
+    tests="${output.summary.totalTests}"
+    time="0">
+`;
+  for (const req of output.requests) {
+    resultXml += `
+    <testsuite
+      name="${req.fileName}"
+      tests="${req.summary.totalTests}"
+      disabled="0"
+      errors="${req.testResults?.filter(r => r.error !== undefined).length}"
+      failures="${req.summary.failedTests}"
+      skipped="0"
+      time="0"
+      timestamp="">
+`;
+    for (const testResult of req.testResults ?? []) {
+      resultXml += `
+      <testcase 
```

Comment:
I am missing a level here. For me, the test file (*.http) but also each individual region would be in this file. Then you could output each test result as <error> or <failure>.

Reply:
In your example on Google Drive, you can actually see what bothers me about it. The properties give the name of the current region. The test case below the properties belongs to this region. I think it would be better encapsulated in a testsuite. How do other reporters do it for jest for example?

Reply:
Sorry I'm not following the concern. I think all the information is present. The testsuite package attribute is used for the file path. The testsuite name is used for the region and the child elements of the testsuite capture the test case names / conditions / assertions. 

What level are we missing here?

Reply:
to answer your question, it looks like jest adopts quite an extensive list of configuration options to help map between jest test conventions and junit xml with as sensible a set of defaults as possible. I think it would be good to try and support some additional configuration as well for this feature -- though perhaps, that can be a follow on merge request.

https://github.com/palmerj3/jest-junit#configuration

Reply:
The properties in your video were at the `<testsuites>` level and therefore also before the `<testsuite>` they belong to. That's what I would have meant. Now I pulled them into testcase and I think they are correct there, even if it is not quite compliant with the XML.

> support some additional configuration as well for this feature -- though perhaps, that can be a follow on merge request.

Yes, let's integrate the feature and then see what else is needed when used. I still spontaneously missing stdout log and possibly the response, but that may simply clutter up the file.